test/sqllogictest/timedomain.slt

```diff
@@ -66,3 +75,69 @@ SELECT * FROM t;
 
 statement ok
 ROLLBACK
+
+# Test that user table and system tables cannot be mixed in a transaction because they
+# belong to different timedomains.
+
+statement ok
+BEGIN;
+
+query I rowsort
+SELECT * FROM t
+----
+
+query error Transactions can only reference objects in the same timedomain.
+SELECT * FROM mz_views
+
+statement ok
+ROLLBACK
+
+# Test that timeline dependent queries can be included in transaction.
+
+statement ok
+BEGIN
+
+query I rowsort
+SELECT * FROM t
+----
+
+query I rowsort
+SELECT mz_now() LIMIT 0
+----
+
+statement ok
+COMMIT
+
+# Test that timeline independent queries can be included at the end transaction.
+
+statement ok
+BEGIN
+
+query I rowsort
+SELECT 1 FROM pg_attribute LIMIT 1
+----
+1
+
+query I rowsort
+SELECT 1 FROM pg_catalog.pg_am LIMIT 1
```

Comment:
So if one day we actually implement this, the test will no longer be testing what we thought it was. If we hard code the current view here, does that still test the same code path, or does it actually need to be a globalid object?

Reply:
Just added those tests